src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -27,29 +39,328 @@ const ActorInit En_Door_InitVars = {
+                // Talon's house door. If Talon has not been woken up at Hyrule Castle
+                // this door should be openable at any time of day. Note that there is no
+                // check for time of day as the scene setup for Lon Lon merely initializes
+                // the door with a different text id between day and night setups
+                doorType = DOOR_NOLOAD;
+            } else {
+                this->actionFunc = EnDoor_WaitForCheck;
+                this->actor.flags |= 0x8000009;
+            }
+        }
+        // Replace the door type it was loaded with by the new type
+        this->actor.params = (this->actor.params & ~0x380) | (doorType << 7);
+    }
+}
+
+#ifdef NON_MATCHING
+// Regalloc only
+void EnDoor_Idle(EnDoor* this, GlobalContext *globalCtx) {
+    Player* player = PLAYER;
+    s32 doorType;
+    Vec3f sp2C;
+    s8 numKeys;
+    s16 phi_v0;
+    s16 temp_v0;
+
+    doorType = (this->actor.params >> 7) & 7;
+    func_8002DBD0(&this->actor, &sp2C, &player->actor.posRot.pos);
+    if (this->unk_191 != 0) {
+        this->actionFunc = EnDoor_Open;
+        SkelAnime_ChangeAnimPlaybackStop(&this->skelAnime, D_809FCECC[this->unk_190], (((s32)player->stateFlags1 * 0x10) < 0) ? 0.75f : 1.5f);
+        if (this->lockTimer != 0) {
+            gSaveContext.dungeonKeys[gSaveContext.mapIndex]--;
+            Flags_SetSwitch(globalCtx, this->actor.params & 0x3F);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHAIN_KEY_UNLOCK);
+        }
+    } else if (!Player_InCsMode(globalCtx)) {
+        if (fabsf(sp2C.y) < 20.0f && fabsf(sp2C.x) < 20.0f && fabsf(sp2C.z) < 50.0f) {
+            phi_v0 = player->actor.shape.rot.y - this->actor.shape.rot.y;
+            if (sp2C.z > 0.0f) {
+                phi_v0 = 0x8000 - phi_v0;
+            }
+            if (ABS(phi_v0) < 0x3000) {
+                if (this->lockTimer != 0) {
+                    numKeys = gSaveContext.dungeonKeys[gSaveContext.mapIndex];
+                    if (numKeys <= 0) {
+                        PLAYER->naviMessageId = -0x203;
+                        return;
+                    } else {
+                        player->doorTimer = 10;
+                    }
+                }
+                player->doorType = (doorType == DOOR_AJAR) ? -1 : 1;
+                player->doorDirection = (sp2C.z >= 0.0f) ? 1.0f : -1.0f;
+                player->doorActor = &this->actor;
+            }
+        } else if (doorType == DOOR_AJAR && this->actor.xzDistFromLink > 2*DOOR_SLAM_RANGE) {
+            // Once the player moves sufficiently far, have the door open slightly again
+            this->actionFunc = EnDoor_IdleAjar;
+        }
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Door/EnDoor_Idle.s")
+#endif
+
+void EnDoor_WaitForCheck(EnDoor* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx)) { // if actor is checked
```

Comment:
"checked"?

Reply:
This actions waits for the initial check before transitioning into `EnDoor_Talk` for the remainder of the check, so checked doesn't really make sense here since `EnDoor_WaitForChecked` implies to me that it waits until the check has ended, which isn't the case.

Reply:
`EnDoor_WaitForTalk` doesn't really work since the A button says `Check` and not `Speak` within the check range. I could rename the talk related names to make them check related instead.

Reply:
I like Talk better, gets the point of using dialogue across much better
And I don't really care about the in-game UI saying Check tbh. As far as code goes, talking to Anju 🐔 or to a door 🚪 or to Talon through a door doesn't make a difference

Reply:
While the text side of the code is identical, the actor itself handles none of that so really the actor should make it clear which of Check and Speak is in use. None of the other internal text machinery features whatsoever so isn't relevant to actor documentation.

Reply:
Well I guess if "check" is supposed to ring the bell of ~"checkable spot" then it would make sense, but since I even started on this by asking what "check" was I guess I won't be convinced about using it

"check" is just way too generic for me